lblod/ember-rdfa-editor
# GN-4387: document language support

### Overview
This PR introduces the concept of a document language. The `doc` node-spec has been modified to support a `lang` attribute. Additionally, a tracked `documentLanguage` setter and getter have been added to the `SayController` class.

This PR comes with some breaking changes:
- The manner in which the `doc` node-spec should be added to an editor schema has been changed. `doc` (imported from `@lblod/ember-rdfa-editor/nodes`) is now a function which accepts an optional `DocumentConfig` object in which it is possible to provide a default language to the `doc` node-spec. 
e.g.:
```js
doc({
  defaultLanguage: 'nl-BE',
}),
```
Note that the defaultLanguage provided here is only used if the parsed document does not already contain a language attribute.
- The behaviour of the `setHtmlContent` method of the `SayController` class has changed. This method now creates a whole new state based on the provided html string. This means plugin state is not carried over from the previous state (e.g. the state of the history plugin resets).

Additionally, this PR adds a new `SetDocAttribute` Prosemirror step. This step allows for updating attributes on the `doc` top-level node in an immutable way.

##### connected issues and PRs:
Jira ticket: https://binnenland.atlassian.net/browse/GN-4387?atlOrigin=eyJpIjoiNTljM2JiN2I0ZDAzNDY0MDk4Njk0MTk4NmIwZTczZDYiLCJwIjoiaiJ9

### How to test/reproduce

- Setting the language of a document can be tested by employing the `documentLanguage` setter on the controller in the Web console by executing e.g. `window.__PC.documentLanguage = 'en-UK';`. If using the Prosemirror-dev-tools, you will notice the `lang` attribute of the top-level `doc` node has changed. 
- In order for a more visual test, you can add the sample `counter` ember-node which should instantaneously react to a document language change.
- The language of the document should be exported when clicking on `Show Export Preview`. The document language should be preserved across reloads of the dummy app.

### Checks PR readiness
